new-frontiers-14/frontier-station-14
# Expedition Console: block mission selection if shuttle has FTL component

## About the PR



Blocks ships from going on expedition if the grid used to send off the ship still has a lingering FTL component.  This fixes an issue where a ship that has returned from expedition gets stuck having "claimed" an expedition but not travelling to it (because the ship is already considered in FTL)

Also, a small fix to the pluralization of "party supliess" - using RobustToolbox's builtin FTL functions, and renaming the punk objective to "party supply stash".

## Why / Balance


Bugfix.

Ideally, the salvage expedition job that gets queued up would notify the console that spawned it that it failed, but this would require larger, more fiddly changes.  As such, bandaid.

## How to test


1. Go on expedition (to make this quicker, edit the values in Content.Shared/CCVar/CCVars.cs and Content.Shared/Salvage/Expeditions/Modifiers/SalvageTimeMod.cs - ideally, there should be frequent refreshes and short expeditions)
2. Open the console as the expedition is ending.
3. When the ship enters FTL, any refresh should not make the buttons clickable, they should remain disabled.
4. When the ship returns from FTL, any refresh should not make the buttons clickable until the FTL component is gone (after the blue circles on the ground disappear)
5. Try to spawn an expedition as soon as you can, it should not claim the button and output a warning saying that your shuttle is already in FTL.

## Media



## Breaking changes


**Changelog**



:cl:
- fix: Expedition ships no longer get stuck when selecting an expedition shortly after returning.
